pages/settings/index.js

```diff
@@ -68,42 +68,39 @@ export default function SettingsPage({
       <StyledTitle>Settings</StyledTitle>
       <StyledSettingsCard>
         <StyledSubheading>Customize Categories</StyledSubheading>
-        <StyledCategoryContainer id="category" name="category">
+        <StyledCategoryContainer>
           {categories.map((category) => (
-            <StyledCategory key={category.id}>
+            <StyledCategory key={category._id}>
               <Image
-                src={getCategoryIcon(category.name)}
+                src={getCategoryIcon(category.name, categories)}
                 alt={`${category.name} icon`}
                 width={36}
                 height={36}
               />
-              <StyledCategoryInput
-                ref={(element) => {
-                  inputRefs.current[category.id] = element;
-                }}
-                type="text"
-                defaultValue={category.name}
-                disabled={category.id !== isEditCategory}
-                autoFocus={category.id === isEditCategory}
-                onChange={(event) => {
-                  if (category.id === isEditCategory) {
-                    category.name = event.target.value;
-                  }
-                }}
-                onKeyDown={(event) => {
-                  if (event.key === "Enter" && category.id === isEditCategory) {
-                    handleSaveEditCategory(category);
-                  }
-                  if (
-                    event.key === "Escape" &&
-                    category.id === isEditCategory
-                  ) {
-                    category.name = originalCategoryName;
-                    event.target.value = originalCategoryName;
-                    event.target.blur();
-                  }
-                }}
-              />
+              {category._id === isEditCategory ? (
+                <StyledCategoryInput
+                  type="text"
+                  defaultValue={category.name}
+                  autoFocus
+                  onKeyDown={(event) => {
+                    if (event.key === "Enter") {
+                      handleSaveEditCategory({
+                        ...category,
+                        name: event.target.value,
+                      });
+                    }
+                    if (event.key === "Escape") {
+                      category.name = originalCategoryName;
+                      event.target.value = originalCategoryName;
```

Comment:
you should not need this, but you should set the isEditCategory to null

Reply:
We need this to reset category name to origin if we abort it with escape .